zerver/lib/markdown/__init__.py

```diff
@@ -1488,6 +1501,27 @@ def content_has_emoji_syntax(content: str) -> bool:
     return re.search(EMOJI_REGEX, content) is not None
 
 
+def get_colon_syntax(text: str) -> str:
+    codepoint = unicode_emoji_to_codepoint(text)
+    if codepoint in codepoint_to_name:
+        colon_emoji = ":" + codepoint_to_name[codepoint] + ":"
+        return colon_emoji
+    return text
+
+
+def translate_unicode_emoji(text: str) -> str:
+    return re.sub(unicode_emoji_regex, lambda emoji: get_colon_syntax(emoji.group("syntax")), text)
+
+
+def content_has_only_emoji_syntax(content: Optional[str]) -> bool:
+    if content is not None:
+        text = translate_emoticons(content.replace("\n", " "))
+        text = translate_unicode_emoji(text)
+        if re.fullmatch(fr" *(?:{EMOJI_REGEX} *)+", text):
+            return True
+    return False
```

Comment:
This is a lot of logic for doing this test.  Can you try using a markdown `PostProcessor` that just checks if the entire rendered message is emoji by inspecting the HTML, and if so, adjusts the markup?

I think if that proves difficult, and maybe if it doesn't, I'd consider not doing anything in the markdown processor and looking at doing that detection `rendered_markdown.js` (aka the client-side logic); both so we don't have two copies and also so we don't clutter the implementation with this much code for such a small feature.